DFC.ServiceTaxonomy.GraphSync/Activities/DeleteContentTypeFromGraphTask.cs

```diff
@@ -0,0 +1,70 @@
+using System.Threading.Tasks;
+using DFC.ServiceTaxonomy.GraphSync.GraphSyncers.Interfaces;
+using Microsoft.AspNetCore.Mvc.Localization;
+using Microsoft.Extensions.Localization;
+using OrchardCore.DisplayManagement.Notify;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Activities;
+using OrchardCore.Workflows.Models;
+
+//todo: part handler called after workflow finishes - can we use that to stop inserts?
+
+namespace DFC.ServiceTaxonomy.GraphSync.Activities
+{
+    public class DeleteContentTypeFromGraphTask : TaskActivity
+    {
+        public DeleteContentTypeFromGraphTask(
+            IDeleteGraphSyncer deleteGraphSyncer,
+            IStringLocalizer<DeleteFromGraphTask> localizer,
+            INotifier notifier)
+        {
+            _deleteGraphSyncer = deleteGraphSyncer;
+            _notifier = notifier;
+            T = localizer;
+        }
+
+        private IStringLocalizer T { get; }
+        private readonly IDeleteGraphSyncer _deleteGraphSyncer;
+        private readonly INotifier _notifier;
+
+        public override string Name => nameof(DeleteContentTypeFromGraphTask);
+        public override LocalizedString DisplayText => T["Delete content type from Neo4j graph"];
+        public override LocalizedString Category => T["Graph"];
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+#pragma warning disable S3220 // Method calls should not resolve ambiguously to overloads with "params"
+            return Outcomes(T["Done"]);
+#pragma warning restore S3220 // Method calls should not resolve ambiguously to overloads with "params"
+        }
+
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            string typeToDelete = GetTypeFromWorkflowContext(workflowContext);
+
+            try
+            {
+                //Delete all nodes by type
+                await _deleteGraphSyncer.DeleteNodesByType(typeToDelete);
+                return Outcomes("Done");
+            }
+            catch (Exception)
+            {
+                _notifier.Add(NotifyType.Error, new LocalizedHtmlString(nameof(DeleteContentTypeFromGraphTask), $"Error: The {typeToDelete} could not be removed because the associated node could not be deleted from the graph. Most likely due to {typeToDelete} having incoming relationships."));
+                throw;
+            }
+        }
+
+        private static string GetTypeFromWorkflowContext(WorkflowExecutionContext workflowContext)
+        {
+            var typeToDelete = workflowContext.Input["ContentType"].ToString();
+
+            if (string.IsNullOrWhiteSpace(typeToDelete))
+                throw new ArgumentNullException($"No Content Type passed to {nameof(DeleteContentTypeFromGraphTask)}");
+
+            return typeToDelete.ToString();
```

Comment:
Looks like it's already a string.

Reply:
It's a nullable string when assigned out of the method without ToString

Reply:
Should be okay, just the type that comes out of the input is nullable and will be caught by the next line anyway.

![image](https://user-images.githubusercontent.com/61147528/76506117-8cbc6f00-6442-11ea-8b43-915cc6cb13a8.png)